REST_AUTHZ_FOR_PLUGINS.md

```diff
@@ -0,0 +1,134 @@
+# Authorization at REST Layer for plugins
```

Comment:
General feedback on this doc. I think it should convey to plugin developers that they can opt not to write a transport action for a route if its unnecessary. From what I understand, plugin developers will write a transport action to protect and endpoint and not because they need to write a transport action. This is fine, but I think the doc should touch on the subject.

Reply:
will add a note here.